src/core/src/core_logic/PatchInstaller.py

```diff
@@ -203,52 +239,43 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
             self.composite_logger.log(progress_status)
 
             # include all dependencies (with specified versions) explicitly
+            # package_and_dependencies initially conains only one package. The dependencies are added in the list by method include_dependencies
             package_and_dependencies = [package]
             package_and_dependency_versions = [version]
-            dependencies = package_manager.get_dependent_list(package)
-            for dependency in dependencies:
-                if dependency not in all_packages:
-                    continue
-                package_and_dependencies.append(dependency)
-                package_and_dependency_versions.append(package_versions[packages.index(dependency)] if dependency in packages else Constants.DEFAULT_UNSPECIFIED_VALUE)
-
-            # multilib resolution for yum
-            if package_manager.get_package_manager_setting(Constants.PKG_MGR_SETTING_IDENTITY) == Constants.YUM:
-                package_name_without_arch = package_manager.get_product_name_without_arch(package)
-                for possible_arch_dependency, possible_arch_dependency_version in zip(packages, package_versions):
-                    if package_manager.get_product_name_without_arch(possible_arch_dependency) == package_name_without_arch and possible_arch_dependency not in package_and_dependencies:
-                        package_and_dependencies.append(possible_arch_dependency)
-                        package_and_dependency_versions.append(possible_arch_dependency_version)
-
-            # remove duplicates
-            package_and_dependencies, package_and_dependency_versions = package_manager.dedupe_update_packages(package_and_dependencies, package_and_dependency_versions)
+            
+            self.include_dependencies(package_manager, [package], all_packages, packages, package_versions, package_and_dependencies, package_and_dependency_versions)
 
             # parent package install (+ dependencies) and parent package result management
             install_result = Constants.FAILED
             for i in range(0, Constants.MAX_INSTALLATION_RETRY_COUNT):
-                install_result = package_manager.install_update_and_dependencies(package_and_dependencies, package_and_dependency_versions, simulate)
+                install_result = package_manager.install_update_and_dependencies_and_get_status(package_and_dependencies, package_and_dependency_versions, simulate)
+
                 if install_result != Constants.INSTALLED:
                     if i < Constants.MAX_INSTALLATION_RETRY_COUNT - 1:
                         time.sleep(i + 1)
                         self.composite_logger.log_warning("Retrying installation of package. [Package={0}]".format(package_manager.get_product_name(package_and_dependencies[0])))
+                else:
+                    break
```

Comment:
Why break out of the loop?

Reply:
There was a bug in existing way of installing packages. The package installation was retried 3 times irrespective of whether the package is installed or not. If the package is installed, then we should not try to install the package again.